### PR TITLE
keep scheduler name consistent as defined in values.yml

### PR DESCRIPTION
**Pull request by onlymellb**
The scheduler can't use scheduler policy configmap correctly when you change the schedulerName in values.yaml. This PR fix it.

### CODE REVIEW

File: charts/tidb-operator/templates/scheduler-deployment.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ spec:
       serviceAccount: {{ .Values.scheduler.serviceAccount }}
     {{- end }}
       containers:
-      - name: tidb-scheduler
+      - name: {{ .Values.scheduler.schedulerName }}
```
Comment (tennix) [inline charts/tidb-operator/templates/scheduler-deployment.yaml]: Container name is used internally, using `tidb-scheduler` is handy for `kubectl exec` and `kubectl logs`. So it's unnecessary to change this.

Comment (onlymellb) [reply charts/tidb-operator/templates/scheduler-deployment.yaml]: I think it's better to keep the name consistent, and it's not necessary to fix the container name to executing `kubectl exec` and `kubectl logs`

Comment (tennix) [reply charts/tidb-operator/templates/scheduler-deployment.yaml]: If using `tidb-scheduler` as container name, the two container names are always `tidb-scheduler` and `kube-scheduler`. When debugging we can always use the same name without thinking what it is. Besides container name should reflect the actual image it uses other than using the installer specified name.

Comment (onlymellb) [reply charts/tidb-operator/templates/scheduler-deployment.yaml]: If you want to use `tidb-scheduler` as container name, you can keep the schedulerName as default. But if you want to change the default value then keeping the container name consistent with it makes sense.

**Pull request merged**